File: src/main/kotlin/com/clean/architecture/cleancepsearcherapi/infraestructure/configuration/AddressConfiguration.java
package com.clean.architecture.cleancepsearcherapi.infraestructure.configuration;

import com.clean.architecture.cleancepsearcherapi.infraestructure.adapters.gateways.SearchAddressGatewayImpl;
import com.clean.architecture.cleancepsearcherapi.adapters.gateways.repositories.AddressRepository;
import com.clean.architecture.cleancepsearcherapi.core.address.usecases.impl.CreationAddressUseCaseImpl;
import com.clean.architecture.cleancepsearcherapi.core.address.usecases.impl.SearchAddressUseCaseImpl;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class AddressConfiguration {

    @Autowired
    private AddressRepository addressRepository;

    @Autowired
    private SearchAddressGatewayImpl searchAddressGateway;

    @Bean
    public CreationAddressUseCaseImpl creationAddressUseCase() {
        return new CreationAddressUseCaseImpl(addressRepository);
    }

    @Bean
    public SearchAddressUseCaseImpl searchAddressUseCase() {
        return new SearchAddressUseCaseImpl(addressRepository, searchAddressGateway);
    }
}

File: src/main/kotlin/com/clean/architecture/cleancepsearcherapi/infraestructure/configuration/AddressConfiguration.kt
package com.clean.architecture.cleancepsearcherapi.infraestructure.configuration

import com.clean.architecture.cleancepsearcherapi.core.ports.AddressRepository
import com.clean.architecture.cleancepsearcherapi.core.address.usecases.impl.CreationAddressUseCaseImpl
import com.clean.architecture.cleancepsearcherapi.core.address.usecases.impl.SearchAddressUseCaseImpl
import com.clean.architecture.cleancepsearcherapi.infraestructure.adapters.gateways.SearchAddressGatewayImpl
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class AddressConfiguration {
    @Autowired
    private val addressRepository: AddressRepository? = null

    @Autowired
    private val searchAddressGateway: SearchAddressGatewayImpl? = null

    @Bean
    fun creationAddressUseCase(): CreationAddressUseCaseImpl {
        return CreationAddressUseCaseImpl(addressRepository!!)
    }

    @Bean
    fun searchAddressUseCase(): SearchAddressUseCaseImpl {
        return SearchAddressUseCaseImpl(addressRepository!!, searchAddressGateway!!)
    }
}